http4k-testing-chaos/src/main/kotlin/org/http4k/chaos/ChaosFilters.kt

```diff
@@ -0,0 +1,34 @@
+package org.http4k.chaos
+
+import org.http4k.core.Filter
+
+const val LATENCY_HEADER = "X-Chaos-Behaviour"
+
+object ChaosFilters {
+    object ChaosPreFilter {
+        operator fun invoke(chaosPolicy: ChaosPolicy, behaviour: ChaosBehaviour): Filter = Filter { next ->
+            { request ->
+                if (chaosPolicy.shouldInject(request)) {
+                    behaviour.inject(request)
+                    next(request).header(LATENCY_HEADER, behaviour.javaClass.name)
```

Comment:
This is no longer LATENCY only - just call it X-http4k-chaos or similar. We can also add the description of the behaviour as a method on the behaviour instance, and then use it to generate dynamic names based on what we did! :)